client/v1/src/main/java/org/opennms/velocloud/client/v1/VelocloudApiCustomerClientV1.java

```diff
@@ -28,50 +28,58 @@
 
 package org.opennms.velocloud.client.v1;
 
+import static org.opennms.velocloud.client.v1.FunctionRefsHolder.ENTERPRISE_GET_ENTERPRISE_EDGES;
+import static org.opennms.velocloud.client.v1.FunctionRefsHolder.ENTERPRISE_GET_ENTERPRISE_USERS;
+import static org.opennms.velocloud.client.v1.FunctionRefsHolder.EVENT_GET_ENTERPRISE_EVENTS;
+import static org.opennms.velocloud.client.v1.FunctionRefsHolder.MONITORING_GET_ENTERPRISE_EDGE_NVS_TUNNEL_STATUS;
+
 import java.time.Instant;
 import java.time.OffsetDateTime;
 import java.time.ZoneId;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Collectors;
 
+import org.opennms.velocloud.client.api.VelocloudApiClientCredentials;
 import org.opennms.velocloud.client.api.VelocloudApiCustomerClient;
 import org.opennms.velocloud.client.api.VelocloudApiException;
-import org.opennms.velocloud.client.api.model.Edge;
 import org.opennms.velocloud.client.api.model.CustomerEvent;
+import org.opennms.velocloud.client.api.model.Edge;
 import org.opennms.velocloud.client.api.model.Link;
 import org.opennms.velocloud.client.api.model.Tunnel;
 import org.opennms.velocloud.client.api.model.User;
-import org.opennms.velocloud.client.v1.api.AllApi;
 import org.opennms.velocloud.client.v1.model.EnterpriseGetEnterpriseEdges;
 import org.opennms.velocloud.client.v1.model.EnterpriseGetEnterpriseUsers;
-
-import org.opennms.velocloud.client.v1.VelocloudApiClientProviderV1.ApiCall;
 import org.opennms.velocloud.client.v1.model.EventGetEnterpriseEvents;
 import org.opennms.velocloud.client.v1.model.EventGetEnterpriseEventsResult;
 import org.opennms.velocloud.client.v1.model.Interval;
 import org.opennms.velocloud.client.v1.model.MonitoringGetEnterpriseEdgeNvsTunnelStatusBody;
 
 public class VelocloudApiCustomerClientV1 implements VelocloudApiCustomerClient {
 
-    private final AllApi api;
+    final ApiExecutor executor;
```

Comment:
Can we bind the credentials/client into some closure that the executor returns? Then we can store the closure here and use it as a replacement for where `AllApi` was used before.